src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -0,0 +1,596 @@
+        }
+
+        final Menu executionMenu = executionTreeNode.getData();
+        if (executionMenu.getMenuType().isNavigationMenu())
+        {
+            throw new CommandException(getCommandUsage(sender, executionTreeNode));
+        }
+
+        final ActionMenu actionMenu = (ActionMenu) executionMenu;
+
+        // Check that required arguments are provided with a valid value
+        final List<ActionArgument> actionArgumentListForActionMenu = actionMenu.getActionArgumentList();
+        for (final ActionArgument actionArgument : actionArgumentListForActionMenu)
+        {
+            if (actionArgument.isRequired())
+            {
+                boolean foundIt = false;
+                if (null != executionActionArgumentList)
+                {
+                    for (final ActionArgument executionActionArgument : executionActionArgumentList)
+                    {
+                        if (null != executionActionArgument)
+                        {
+                            if (actionArgument.getName().equals(executionActionArgument.getName()))
+                            {
+                                if (!executionActionArgument.isValueSet())
+                                {
+                                    if (null == badArgument)
+                                    {
+                                        throw new CommandException(
+                                                getCommandUsage(sender, executionTreeNode)
+                                                    + ": no value specified for required argument " + actionArgument.getName());
+                                    }
+                                    else
+                                    {
+                                        throw new CommandException(
+                                                getCommandUsage(sender, executionTreeNode)
+                                                    + ": invalid value '" + badArgument + "' for required argument " + actionArgument.getName());
+                                    }
+                                }
+                                else
+                                {
+                                    foundIt = true;
+                                }
+                            }
+                        }
+                    }
+                }
+                if (!foundIt)
+                {
+                    throw new CommandException(getCommandUsage(sender, executionTreeNode)
+                            + ": missing required parameter " + actionArgument.getName());
+                }
+            }
+        }
+
+        if (null != executionActionArgumentList)
+        {
+            for (final ActionArgument executionActionArgument : executionActionArgumentList)
+            {
+                if (null != executionActionArgument)
+                {
+                    if (!executionActionArgument.isValueSet())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)

Reply:
this